libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts

```diff
@@ -0,0 +1,29 @@
+import { Resolver, Query } from '@nestjs/graphql'
+import { NationalRegistryXRoadService } from './nationalRegistryXRoad.service'
+import {
+  CurrentUser,
+  IdsAuthGuard,
+  IdsUserGuard,
+  ScopesGuard,
+  User,
+} from '@island.is/auth-nest-tools'
+import { UseGuards } from '@nestjs/common'
+import { NationalRegistryPerson } from '../models/nationalRegistryPerson.model'
+
+@UseGuards(IdsAuthGuard, IdsUserGuard, ScopesGuard)
+@Resolver()
+export class NationalRegistryXRoadResolver {
+  constructor(
+    private nationalRegistryXRoadService: NationalRegistryXRoadService,
+  ) {}
+
+  @Query(() => NationalRegistryPerson, { nullable: true })
+  async getChildrenCustodyAndParents(
```

Comment:
This is looking a lot better!

The only thing that's missing is using the power of GraphQL to only fetch specific fields when requested... here if you make the resolver a resolver for the `nationalRegistryPerson`, you can use decorators such as ResolveField to explicitly fetch the desired fields.
Steps to do that:
1. @Resolver(() => NationalRegistryPerson)
2. Rename query from `getChildrenCustodyAndParents` to `nationalRegistryPersons` (don't use get.* to follow the graphql naming conventions)
3. Rename `this.nationalRegistryXRoadService.getCustodyChildrenAndParents` to `this.nationalRegistryXRoadService.getNationalRegistryPerson` and only return `einstaklingarGetEinstaklingur`
4. Make `children` a `ResolveField` which explicitly calls `this.nationalRegistryXRoadService.getCustodyChildren` where the rest of the API calls are made


Here you save a ton of requests when it's not needed. Let's say a client only wants to fetch the address, or the fullName:
```graphql
query nationalRegistryUser {
  nationalId
  fullName
}
```
now only a single API call is made, since `children` wasn't explicitly asked for => which I call the power of GraphQL :D

Reply:
Niceeee